docs/program-setup/bare_metal/mbed2_porting.md

```diff
@@ -0,0 +1,125 @@
+
+The main steps are:
+1. Clone [mbed-os-example-blinky-baremetal](https://github.com/ARMmbed/mbed-os-example-blinky-baremetal).
+1. Edit `targets.json`. In this step, you will configure your target to support Mbed OS 6 and add bare metal configuration parameters.
+1. Build and run mbed-os-example-blinky-baremetal. In this step, you will validate the configuration and troubleshoot any issue.
+
+### Clone mbed-os-example-blinky-baremetal
+
+Run the following command:
+- `git clone https://github.com/ARMmbed/mbed-os-example-blinky-baremetal`
+
+And then change directory:
+- `cd mbed-os-example-blinky-baremetal`
+
+### Edit targets.json
+
+Find your target in `mbed-os/targets/targets.json` and update its configuration as described below.
+
+Configure your target to support Mbed OS 6:
+- Remove the `release_versions` property as it is no longer required.
+```
+"release_versions": ["2"]
+```
+
+- Add the `supported_application_profiles` property to indicate that the application profile supported by this target is bare metal.
+```
+"supported_application_profiles" : ["bare-metal"]
+```
+
+- Override `supported_c_libs` property to link with the smaller C libraries. The default for all targets is defined as follows: 
+```
+"supported_c_libs": {
+    "arm":  ["std"],
+    "gcc_arm":  ["std", "small"],
+    "iar": ["std"]
+}
+```
+
+Both the ARM and GCC_ARM toolchains support optimized versions of the C standard libraries, microlib and newlib-nano respectively. We recommend using them with the bare metal profile to achieve lower memory footprints. Ultra constrained targets should override `supported_c_libs` as follows:
+```
+"supported_c_libs": {
+    "arm":  ["small"],
+    "gcc_arm":  ["small"]
+}
+```
+
+For each toolchain, if there is enough memory to link with the standard library, add the corresponding "std" library to the list. For example:
+```
+"supported_c_libs": {
+    "arm":  ["std", "small"],
+    "gcc_arm":  ["std", "small"],
+    "iar": ["std"]
+}
+```
+<span class="notes">**Note:** For ARM, your target scatter file needs to have a two-region memory model. If the build system throws an error (presence of undefined symbols `Image$$ARM_LIB_HEAP$$ZI$$Base`, `Image$$ARM_LIB_HEAP$$ZI$$Length`, `Image$$ARM_LIB_HEAP$$ZI$$Limit`) when compiling with microlib, you can find more information [here](https://os.mbed.com/docs/mbed-os/v6.0-preview/reference/using-small-c-libraries.html).</span>
+
+<span class="notes">**Note:** The IAR toolchain does not have a small C library.</span>
+
+- Set the `c_lib` property to indicate which C library should be used by the build tools. If your target has `"default_lib": "small"` defined, then replace it with `"c_lib" : "small"`. Otherwise add `"c_lib" : "small"`. We recommend that ultra constrained devices running bare metal applications link with the small C libraries by default.
```

Comment:
```suggestion
- We recommend that ultraconstrained devices running bare metal applications link with the small C libraries by default. To indicate which C library should be used by the build tools, set the `c_lib` property:
    - If your target has `"default_lib": "small"` defined, then replace it with `"c_lib" : "small"`. 
    - Otherwise, add `"c_lib" : "small"`. 
```